src/AppContainer.tsx

```diff
@@ -0,0 +1,322 @@
+
+const USER_ID = 12121;
+
+export const AppContainer: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState(Filters.All);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isPageLoading, setIsPageLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isUpdatingId, setIsUpdatingId] = useState<number[]>([]);
+  const [isEditing, setIsEditing] = useState<number | null>(null);
+  const [editQuery, setEditQuery] = useState('');
+  const [editQueryPrev, setEditQueryPrev] = useState('');
+
+  const handleErrorMessage = (message: string) => {
+    setErrorMessage(message);
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  };
+
+  const closeError = () => {
+    setErrorMessage('');
+  };
+
+  const loadTodos = useCallback(async () => {
+    try {
+      const loadedTodos = await getTodos(USER_ID);
+
+      setTodos(loadedTodos);
+      setIsPageLoading(true);
+    } catch {
+      handleErrorMessage(ErrorMessage.loadTodo);
+    }
+  }, []);
+
+  useEffect(() => {
+    loadTodos();
+  }, [loadTodos]);
+
+  const visibleTodos = (filterBy: Filters) => {
+    if (!todos) {
+      return todos;
+    }
+
+    switch (filterBy) {
+      case Filters.All:
+        return todos;
+      case Filters.Active:
+        return todos.filter(todo => !todo.completed);
+      case Filters.Completed:
+        return todos.filter(todo => todo.completed);
+      default:
+        return todos;
+    }
+  };
+
+  const handleFormSubmit = async (e?: Event) => {
+    e?.preventDefault();
+    if (!query.trim().length) {
+      handleErrorMessage(ErrorMessage.titleEmpty);
+
+      return;
+    }
+
+    try {
+      const newTodo = {
+        userId: USER_ID,
+        title: query.trim(),
+        completed: false,
+        id: 0,
+      };
+
+      setTempTodo({ ...newTodo, id: 0 });
+      setIsPageLoading(false);
+
+      const newTodos = await addTodo('/todos', newTodo);
+
+      setTodos(state => {
+        return [...state, newTodos as Todo];
+      });
+
+      setTempTodo(null);
+      setIsPageLoading(true);
+      setQuery('');
+    } catch {
+      handleErrorMessage(ErrorMessage.addTodo);
+      setTempTodo(null);
+      setIsPageLoading(true);
+    }
+  };
+
+  const handleComplete = async (id: number, completed: boolean) => {
+    try {
+      setIsUpdatingId((state) => [...state, id]);
+      const updatedData: Todo = {
+        id,
+        userId: USER_ID,
+        title: query.trim(),
+        completed: !completed,
+      };
+
+      await updateTodo(id, updatedData);
+      setTodos((state) => state.map((todo) => {
+        if (todo.id === id) {
+          return {
+            ...todo,
+            completed: !completed,
+          };
+        }
+
+        return todo;
+      }));
+      setIsUpdatingId((state) => state.filter((stateId) => stateId !== id));
```

Comment:
Add some empty lines to your code. Currently it looks to dense to mee
```suggestion
  const handleComplete = async (id: number, completed: boolean) => {
    try {
      setIsUpdatingId((state) => [...state, id]);
  
      const updatedData: Todo = {
        id,
        userId: USER_ID,
        title: query.trim(),
        completed: !completed,
      };

      await updateTodo(id, updatedData);
   
      setTodos((state) => state.map((todo) => {
        if (todo.id === id) {
          return {
            ...todo,
            completed: !completed,
          };
        }

        return todo;
      }));
     
      setIsUpdatingId((state) => state.filter((stateId) => stateId !== id));
```